src/java/org/apache/cassandra/index/sai/memory/RowMapping.java

```diff
@@ -64,7 +64,7 @@ public int size()
         }
     };
 
-    private final InMemoryTrie<Integer> rowMapping = InMemoryTrie.shortLived();
+    private final InMemoryTrie<Integer> rowMapping = InMemoryTrie.shortLived(ByteComparable.Version.OSS41);
```

Comment:
I'm not quite sure where the best place for it is, but it would be nice to do like for the memtable tries and add a constant somewhere for "the SAI default byte comparable version" and use that here and elsewhere, instead of use `OSS41` each time.

Maybe one option would be to add a `defaultByteComparableVersion()` method to `org.apache.cassandra.index.sai.disk.format.Version`, have that default to `OSS41` for now, and call `Version.latest().defaultByteComparableVersion()` where we need it. I'll note that there is already a `Version.byteComparableVersionFor` method that is use to know the version for a file we read, hence the suggestion.

Reply:
Putting it in `Version` would make readers believe its version-specific, while making it so would require quite a few changes that I don't feel equipped to do right now.

Finding the right place for it is tricky. I chose `TypeUtil` because it holds methods related to the conversion and types in general. I was also considering `StorageAttachedIndex`; if you have a better idea where to put it, let me know.